### PR TITLE
Fixes mirage/ocaml-cohttp#218

### PR DESCRIPTION
**Pull request by marklrh**
Use Cmdliner for CLI

### CODE REVIEW

File: bin/cohttp_server_lwt.ml [hunk 1]
```diff
@@ -147,33 +147,52 @@ let start_server docroot port host index verbose cert key () =
...
+  let doc = "TCP port to listen on" in
+  Arg.(value & opt int 8080 & info ["p"] ~docv:"PORT" ~doc)
+
+let index =
+  let doc = "Name of index file in directory" in
+  Arg.(value & opt string "index.html" & info ["i"] ~docv:"INDEX" ~doc)
+
+let verb =
+  let doc = "logging output to console" in
+  Arg.(value & flag & info ["v"; "verbose"] ~doc)
```
Comment (dbuenzli) [inline bin/cohttp_server_lwt.ml]: Doc string starting with a lowercase. Also I do put a dot at the end of doc strings but you are consistent in not doing so.

File: bin/cohttp_server_lwt.ml [hunk 2]
```diff
@@ -147,33 +147,52 @@ let start_server docroot port host index verbose cert key () =
...
+  let doc = "SSL certificate file" in
+  Arg.(value & opt (some string) None & info ["c"] ~docv:"SSL_CERT" ~doc)
+
+let ssl_key =
+  let doc = "SSL key file" in
+  Arg.(value & opt (some string) None & info ["k"] ~docv:"SSL_KEY" ~doc)
+
+let doc_root = 
+  let doc = "serving directory" in
+  Arg.(value & pos 0 dir "." & info [] ~docv:"DOCROOT" ~doc)
```
Comment (dbuenzli) [inline bin/cohttp_server_lwt.ml]: Doc string starting with a lowercase.

File: bin/cohttp_server_lwt.ml [hunk 3]
```diff
@@ -147,33 +147,52 @@ let start_server docroot port host index verbose cert key () =
...
+
+let cmd =
+  let doc = "a simple http server" in
+  let man = [
+    `S "DESCRIPTION";
+    `P "$(tname) sets up a simple http server with lwt as backend";
+    `S "BUGS";
+    `P "Report them to via e-mail to <mirageos-devel@lists.xenproject.org>, or
+        on the issue tracker at <https://github.com/mirage/ocaml-cohttp/issues>";
```
Comment (dbuenzli) [inline bin/cohttp_server_lwt.ml]: Report them ~~to~~ via email to

File: bin/cohttp_server_lwt.ml (hunk 1 above)
Comment (marklrh) [inline bin/cohttp_server_lwt.ml]: Yeah, I should add a dot. I used [this program](https://github.com/mirage/ocaml-cohttp/blob/master/bin/cohttp_curl_lwt.ml#L67) as my reference. It uses uppercase instead. I am not sure now

Comment (marklrh) [inline bin/cohttp_server_lwt.ml]: @dbuenzli some word like "SSL" starts with uppercase letter. Should I change it to lowercase instead?

Comment (marklrh) [inline bin/cohttp_server_lwt.ml]: @dbuenzli ouch I see. You mean I should not let docstring start with lowercase? gotta ya

**Comment by rgrinberg** [comment]
@marklrh thank you!

**Pull request merged**